lock-api/src/main/java/com/palantir/lock/watch/NoOpLockWatchEventCache.java

```diff
@@ -55,7 +53,11 @@ public CommitUpdate getCommitUpdate(long startTs) {
     @Override
     public TransactionsLockWatchEvents getEventsForTransactions(Set<Long> startTimestamps,
             Optional<IdentifiedVersion> version) {
-        return NONE;
+        return ImmutableTransactionsLockWatchEvents.builder()
+                .clearCache(true)
+                .startTsToSequence(
+                        startTimestamps.stream().collect(Collectors.toMap(startTs -> startTs, $ -> FAKE_VERSION)))
```

Comment:
Nice! For extra safety, I would probably generate a new UUID every time, but I think this will translate to a snapshot and a clear cache for each transaction that starts this way.